app/models/concerns/product_stock.rb

```diff
@@ -0,0 +1,25 @@
+require 'active_support/concern'
+
+module ProductStock
+  extend ActiveSupport::Concern
+
+  def on_demand
+    if has_variants?
+      if variants.size > 1
+        raise 'Cannot determine product on_demand value of product with multiple variants'
+      else
+        variants.first.on_demand
+      end
+    else
+      master.on_demand
+    end
+  end
+
+  def on_hand
+    if has_variants?
+      variants.map(&:on_hand).reduce(:+)
```

Comment:
I think AR's [sum](https://apidock.com/rails/v3.2.13/ActiveRecord/Calculations/sum) would be more performant as we wouldn't load all the variant objects, iterate to get their on_hand value and then sum them all. Something like:

```suggestion
      variants.sum(:on_hand)
```

Reply:
I will do that.
this is your code btw ;-)
https://github.com/openfoodfoundation/openfoodnetwork/commit/75ad4254a6c50b28a2f3b36828d2e3428b20b774#diff-e6a3a7e1c82ac34e5e72899c9ceb109cR133

Reply:
Yes, I saw. I was probably too lazy

Reply:
I don't think ActiveRecord's `sum` works here, because `on_hand` is not a column in the database, right? Each variant actually needs to find all stock locations first. And the variants may be augmented by variant overrides.

Reply:
hey @mkllnk , thanks, that makes sense!

Reply:
I'll skip this. ok?

Reply:
@mkllnk my point here was to get the sum straight from DB; way faster than Ruby.

Reply:
Sure. :-)